Sources/AWSSDKSwiftCore/Encoder/XMLDecoder.swift

```diff
@@ -224,21 +230,22 @@ fileprivate class _XMLDecoder : Decoder {
                 // if entry name is NULL, set enclosing xml element to be the parent element and set the name to look for to be the name of the current element. Otherwise the code will look for xml elements named entryName under the current element
                 if entryName == nil {
                     entryName = decoder.codingPath.last?.stringValue
-                    if let parent = element.parent as? XML.Element {
+                    if let parent = element.parent {
                         decoder.storage.popContainer()
                         decoder.storage.push(container: parent)
                         element = parent
                     }
                 }
                 if let entryName = entryName {
-                    let entries = element.elements(forName: entryName)
-                    for entry in entries {
-                        if let keyElement = entry.child(for: keyName), let valueElement = entry.child(for: valueName) {
-                            guard let keyString = keyElement.stringValue else { continue }
-                            if let key = Key(stringValue: keyString) {
-                                allKeys.append(key)
-                                // store value elements for later
-                                allValueElements[keyString] = valueElement
+                    if let entries = (element as? XML.Element)?.elements(forName: entryName) {
+                        for entry in entries {
+                            if let keyNode = entry.child(for: keyName), let valueNode = entry.child(for: valueName) {
+                                guard let keyString = keyNode.stringValue else { continue }
+                                if let key = Key(stringValue: keyString) {
```

Comment:
Maybe combine `if` and `guard` and `if`?
```swift
guard let keyNode = entry.child(for: keyName), 
      let valueNode = entry.child(for: valueName), 
      let keyString = keyNode.stringValue, 
      let key = Key(stringValue: keyString) else 
{
  continue
}
allKeys.append(key)
// store value elements for later
allValueNodes[keyString] = valueNode
```

IMHO this would make reading way easier!

Reply:
These all make sense